graphrag_sdk/models/litellm.py

```diff
@@ -0,0 +1,283 @@
+            ),
+            system_instruction=json["system_instruction"],
+        )
+
+
+class LiteModelChatSession(GenerativeModelChatSession):
+    """
+    A chat session for interacting with the LiteLLM model, maintaining conversation history.
+    """
+
+    def __init__(self, model: LiteModel, args: Optional[dict] = None):
+        """
+        Initialize the chat session and set up the conversation history.
+
+        Args:
+            model (LiteLLMGenerativeModel): The model instance for the session.
+            args (Optional[dict]): Additional arguments for customization.
+        """
+        self._model = model
+        self._args = args
+        self._chat_history = (
+            [{"role": "system", "content": self._model.system_instruction}]
+            if self._model.system_instruction is not None
+            else []
+        )
+        
+    def get_chat_history(self) -> list[dict]:
+        """
+        Retrieve the conversation history for the current chat session.
+
+        Returns:
+            list[dict]: The chat session's conversation history.
+        """
+        return self._chat_history.copy()
+
+    def send_message(self, message: str, output_method: OutputMethod = OutputMethod.DEFAULT) -> GenerationResponse:
+        """
+        Send a message in the chat session and receive the model's response.
+
+        Args:
+            message (str): The message to send.
+            output_method (OutputMethod): Format for the model's output.
+
+        Returns:
+            GenerationResponse: The generated response.
+        """
+        generation_config = self._adjust_generation_config(output_method)
+        self._chat_history.append({"role": "user", "content": message})
+        try:
+            response = completion(
+                model=self._model.model,
+                messages=self._chat_history,
+                **generation_config
+            )
+        except Exception as e:
+            # Handle exception (e.g., log error, retry, or raise a custom exception)
+            raise RuntimeError("Error during completion request") from e
+        content = self._model.parse_generate_content_response(response)
+        self._chat_history.append({"role": "assistant", "content": content.text})
+        return content
+    
+    def _adjust_generation_config(self, output_method: OutputMethod):
+        """
+        Adjust the generation configuration based on the specified output method.
+
+        Args:
+            output_method (OutputMethod): The desired output method (e.g., default or JSON).
+
+        Returns:
+            dict: The adjusted configuration settings for generation.
+        """
+        config = self._model.generation_config.to_json()
+        if output_method == OutputMethod.JSON:
+            config['temperature'] = 0
+            config['response_format'] = { "type": "json_object" }
+        
+        return config
+    
+    def delete_last_message(self):
```

Comment:
Remind me, why do we need this function?

Reply:
to clear bad Cypher generation from the history